packages/node_modules/@webex/plugin-meetings/src/meeting/audio.js

```diff
@@ -0,0 +1,99 @@
+          from: '*',
+          to: function (audio, meeting) {
+            return handleTransition(this.state, audio, meeting);
+          }
+        },
+        {
+          name: MEETING_AUDIO_STATE_MACHINE.TRANSITIONS.TOGGLE,
+          from: '*',
+          to: function (audio, meeting) {
+            return handleTransition(this.state, audio, meeting);
+          }
+        }
+      ],
+      data: {
+        muted: undefined,
+        self: undefined
+      },
+      methods: {
+        getMuted() {
+          return this.muted;
+        },
+        getSelf() {
+          return this.self;
+        },
+        onAfterInit(transition, audio, meeting) {
+          this.muted = audio.mute;
+          this.self = audio.self;
+          meeting.trigger(EVENT_TRIGGERS.MEDIA_AUDIO, {
+            type: EVENT_TYPES.local,
+            status: this.state,
+            muted: this.muted,
+            action: MEETING_AUDIO_STATE_MACHINE.TRANSITIONS.INIT
+          });
+          console.log(`audio:state#onAfterInit->${transition.event} fired! State changed from '${transition.from}' to '${transition.to}' with transition '${transition.transition}''.`);
+        },
+        onAfterHistoryBack(transition, meeting, error) {
+          meeting.trigger(EVENT_TRIGGERS.MEDIA_AUDIO, {
+            type: EVENT_TYPES.ERROR,
+            status: this.state,
+            muted: this.muted,
+            action: MEETING_AUDIO_STATE_MACHINE.TRANSITIONS.TOGGLE,
+            error
+          });
+          console.log(`audio:state#onAfterHistoryBack->${transition.event} fired! State failed to change with transition '${transition.transition}''. State has been reset to ${this.history[this.history.length - 1]} and toggling has been reversed.`);
+        },
+        onAfterToggle(transition, audio, meeting) {
+          if (audio.self) {
+            this.self = true;
+          }
+          if (transition.from !== transition.to) {
+            MeetingUtil.remoteUpdateAudioVideo(audio, {mute: meeting.video.muted}, meeting)
+              .then((resolution) => {
+                Media.localTracks(audio, {mute: meeting.video.muted}, meeting.localMediaStream);
+                this.muted = audio.mute;
+                meeting.trigger(EVENT_TRIGGERS.MEDIA_AUDIO, {
+                  type: EVENT_TYPES.local,
+                  status: this.state,
+                  muted: this.muted,
+                  action: MEETING_AUDIO_STATE_MACHINE.TRANSITIONS.TOGGLE,
+                  resolution
+                });
+                console.log(`audio:state#onAfterToggle->${transition.event} fired! State changed from '${transition.from}' to '${transition.to}' with transition '${transition.transition}''.`);
+              }).catch((err) => {
+                this.historyBack(meeting, err);
+              });
+          }
+        },
+        plugins: [
+          new StateMachineHistory({max: 2})
+        ]
+      }
+    });
+  }
+};
+
+export default AudioStateMachine;
```

Comment:
This feels really complex for a simple binary property. @mattnorris?

Reply:
@andrewholsted my understanding from talking with @chrisadubois is that it will allow us to recover from a myriad things happening as these complex calls occur. That's oversimplifying it, and we can discuss, but I believe this exposes the life cycles Chris wants to be able to reverse history and recover from errors. 

@chrisadubois can you elaborate?

Reply:
that's exactly correct matt, as discussed from the last meeting.

And the properties this.audio.muted will always have a source of truth. for details one can dig into the state